.github/workflows/ghc-lib-runhaskell-ghc-9.10.1.yml

```diff
@@ -11,33 +11,35 @@ jobs:
         os: [ubuntu, macos, windows]
     steps:
       - uses: actions/checkout@v2
-      - uses: haskell-actions/setup@v2
-        with:
-          enable-stack: true
-          stack-version: '2.15.7'
+      - run: |-
+          curl -sSL https://get.haskellstack.org/ | sh -s - -f
+          curl --proto '=https' --tlsv1.2 -sSf https://get-ghcup.haskell.org | BOOTSTRAP_HASKELL_NONINTERACTIVE=1 BOOTSTRAP_HASKELL_GHC_VERSION=9.10.1 BOOTSTRAP_HASKELL_CABAL_VERSION=latest BOOTSTRAP_HASKELL_INSTALL_NO_STACK=1 BOOTSTRAP_HASKELL_INSTALL_NO_STACK_HOOK=1 BOOTSTRAP_HASKELL_ADJUST_BASHRC=P sh
+        shell: bash
```

Comment:
i need more deterministic control  over what gets installed where than i get leaving it to the actions. the version installed is latest (2.15.7). i know at all times what is getting installed here.

Reply:
Seems reasonable - though i worry that when a new stack version comes out, this will update without us checking, and could break unexpectedly. Is there a way to tell `https://get.haskellstack.org/` to explicitly install `2.15.7`?

Reply:
i don't know - probably (and if i come across a way/the way to do that sure i'll put up a change) but honestly, there is just no way this will happen (the builds breaking whether its stack related or something else) without me getting signal. i have automated jobs running every day.

Reply:
generally speaking by the way we want stack latest (whatever that is) and to get signal if using stack latest doesn't work. note also that in recent revisions of these actions up to the release of 2.15.7  we've been installing stack only to then use it to build itself from git (because stack latest wouldn't work with ghc-9.10.1). once again, let me assure you once again, that though your concern is appreciated, you shouldn't be worried here.

Reply:
Alright, thanks for the thorough explanation

Reply:
i didn't even get started samuel 😁 i could speak for days about all the esoteric details i've tragically had to discover that relate to this topic!

Reply:
I can only imagine! Appreciate you handling them :)